engine/runner/src/main/java/org/enso/runner/Main.java

```diff
@@ -0,0 +1,1193 @@
+   */
+  private static void run(
+      String path,
+      java.util.List<String> additionalArgs,
+      scala.Option<String> projectPath,
+      Level logLevel,
+      boolean logMasking,
+      boolean enableIrCaches,
+      boolean disablePrivateCheck,
+      boolean enableAutoParallelism,
+      boolean inspect,
+      boolean dump,
+      scala.Option<String> executionEnvironment,
+      int warningsLimit)
+      throws IOException {
+    var fileAndProject = Utils.findFileAndProject(path, projectPath.getOrElse(() -> null));
+    if (fileAndProject == null) {
+      throw exitFail();
+    }
+    var projectMode = fileAndProject._1();
+    var file = fileAndProject._2();
+    var projectRoot = fileAndProject._3();
+    var options = new HashMap<String, String>();
+    if (dump) {
+      options.put("engine.TraceCompilation", "true");
+      options.put("engine.MultiTier", "false");
+    }
+    if (inspect) {
+      options.put("inspect", "");
+    }
+    var context =
+        ContextFactory.create()
+            .projectRoot(projectRoot)
+            .in(System.in)
+            .out(System.out)
+            .repl(new Repl(makeTerminalForRepl()))
+            .logLevel(logLevel)
+            .logMasking(logMasking)
+            .enableIrCaches(enableIrCaches)
+            .disablePrivateCheck(disablePrivateCheck)
+            .strictErrors(true)
+            .enableAutoParallelism(enableAutoParallelism)
+            .executionEnvironment(
+                executionEnvironment.isDefined() ? executionEnvironment.get() : null)
+            .warningsLimit(warningsLimit)
+            .options(options)
+            .build();
+
+    if (projectMode) {
+      var result = PackageManager$.MODULE$.Default().loadPackage(file);
+      if (result.isSuccess()) {
+        var s = (Success) result;
+        @SuppressWarnings("unchecked")
+        var pkg = (org.enso.pkg.Package<java.io.File>) s.get();
+        var main = pkg.mainFile();
+        if (!main.exists()) {
+          println("Main file does not exist.");
+          context.context().close();
+          throw exitFail();
+        }
+        var mainModuleName = pkg.moduleNameForFile(pkg.mainFile()).toString();
+        runPackage(context, mainModuleName, file, additionalArgs);
+      } else {
+        println(((Failure) result).exception().getMessage());
+        throw exitFail();
+      }
+    } else {
+      runSingleFile(context, file, additionalArgs);
+    }
+    context.context().close();
+    throw exitSuccess();
+  }
+
+  /**
+   * Handles the `--docs` CLI option.
+   *
+   * <p>Generates reference website from standard library.
+   *
+   * @param projectPath if specified, the docs is generated for a project at the given path
+   * @param logLevel log level to set for the engine runtime
+   * @param logMasking is the log masking enabled
+   * @param enableIrCaches are the IR caches enabled
+   */
+  private static void genDocs(
+      scala.Option<String> projectPath,
```

Comment:
Can we get rid of all these `scala.Option` usages? At least in private methods? Replace it either with `java.util.Optional`, or maybe even better, just a plain `String` with `null`.

Reply:
We should get right of the `scala.*` classes. So far I was trying to not change the semantics, so I left them in place.

Reply:
It is hard to get rid of `scala.Option` as new instances of Scala classes created by `Main` seem to require it. However I at least remove the `Either` equilibristic in 1d17383